From 81f595adf46f37ec05782e79b6e00947972163a4 Mon Sep 17 00:00:00 2001
From: Eelco Dolstra <eelco.dolstra@logicblox.com>
Date: Fri, 12 Apr 2013 13:16:57 +0200
Subject: [PATCH] Don't try to unmount /nix or /nix/store

They'll still be remounted read-only.

https://github.com/NixOS/nixos/issues/126
Fix mount option `x-initrd.mount` handling (#35268) (#16)
---
 src/core/mount.c      | 4 +++-
 src/shutdown/umount.c | 2 ++
 2 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/core/mount.c b/src/core/mount.c
index 959b8fbed2..6bc3569550 100644
--- a/src/core/mount.c
+++ b/src/core/mount.c
@@ -409,7 +409,9 @@ static bool mount_is_extrinsic(Mount *m) {
 
         if (PATH_IN_SET(m->where,  /* Don't bother with the OS data itself */
                         "/",
-                        "/usr"))
+                        "/usr",
+                        "/nix/",
+                        "/nix/store"))
                 return true;
 
         if (PATH_STARTSWITH_SET(m->where,
diff --git a/src/shutdown/umount.c b/src/shutdown/umount.c
index c75be39fef..ccc0e26819 100644
--- a/src/shutdown/umount.c
+++ b/src/shutdown/umount.c
@@ -371,6 +371,8 @@ static int delete_dm(dev_t devnum) {
 
 static bool nonunmountable_path(const char *path) {
         return path_equal(path, "/")
+                || path_equal(path, "/nix")
+                || path_equal(path, "/nix/store")
 #if ! HAVE_SPLIT_USR
                 || path_equal(path, "/usr")
 #endif
-- 
2.23.0

